### ISSUE TITLE
Fully support ghost vertices on edge and chain shapes

### ISSUE THREAD

**Issue by slime73** [love2d/love#1196]
**[Original report](https://bitbucket.org/rude/love/issue/1191) by airstruck (Bitbucket: [airstruck](https://bitbucket.org/airstruck), GitHub: [airstruck](https://github.com/airstruck)).**

----------------------------------------

Edge shapes currently have no support for ghost vertices. Chain shapes only have partial support for ghost vertices (they can be created, but not retrieved).

Page 19 of the Box2D manual gives an example of ghost vertices on edge shapes:

```
#!c++

// This is an edge shape with ghost vertices.
b2Vec2 v0(1.7f, 0.0f);
b2Vec2 v1(1.0f, 0.25f);
b2Vec2 v2(0.0f, 0.0f);
b2Vec2 v3(-1.7f, 0.4f);
b2EdgeShape edge;
edge.Set(v1, v2);
edge.m_hasVertex0 = true;
edge.m_hasVertex3 = true;
edge.m_vertex0 = v0;
edge.m_vertex3 = v3;
```

Public fields `m_hasVertex0` and `m_vertex0` store the previous ghost vertex, and `m_hasVertex3` and `m_vertex3` store the next ghost vertex. These could be exposed through two functions matching Love's chain shape API, `setPreviousVertex` and `setNextVertex`, and two functions to retrieve those values, `getPreviousVertex` and `getNextVertex`.

Chain shapes should also be able to retrieve ghost vertices, e.g. for serialization and debug drawing, via new `getPreviousVertex` and `getNextVertex` functions. Box2D stores these values in the following public fields:

```
#!c++

b2Vec2 m_prevVertex, m_nextVertex;
bool m_hasPrevVertex, m_hasNextVertex;
```

In summary, this ticket requests:

- 4 new methods for EdgeShape: `setPreviousVertex`, `setNextVertex`, `getPreviousVertex`, and `getNextVertex`

- 2 new methods for ChainShape: `getPreviousVertex` and `getNextVertex`
